### PR TITLE
chat: add chat connection CHA-CS* spec points

### PR DESCRIPTION
**Pull request by AndyTWF**
This change adds the spec points for chat connections, CHA-CS*.

### CODE REVIEW

File: textile/chat-features.textile [hunk 1]
```diff
@@ -34,6 +34,36 @@ h3(#rooms-general). General Information
 
 A Room is the atomic unit of Chat. All chat operations are performed in the context of a room that a client is currently attached to. As the Room is the atomic unit, its state should be considered as the combination of the states of underlying resources.
 
+h3(#connection-status). Connection Status
+
+The connection status of the Chat client closely reflects the status of the underlying Realtime client, but with one major difference. When a disconnect takes place, we assume that it is transient for a certain amount of time, before eventually emitting an event. This prevents customers being unnecessarily sent connection status updates if the connection recovers quickly (e.g. when going through a tunnel).
+
+* @(CHA-CS1)@ Every chat client has a @status@, which describes the current status of the connection.
+** @(CHA-CS1a)@ The @INITIALIZED@ status is a default status when the realtime client is first initialized. This value will only (likely) be seen if the realtime client doesn't have autoconnect turned on.
```
Comment (splindsay-92) [inline textile/chat-features.textile:42]:
```suggestion
** @(CHA-CS1a)@ The @INITIALIZED@ status is a default status when the realtime client is first initialized. This value may be seen if the realtime client doesn't have autoconnect turned on.
```
or maybe
```suggestion
** @(CHA-CS1a)@ The @INITIALIZED@ status is a default status when the realtime client is first initialized. This value shall (in most cases) be seen if the realtime client doesn't have autoconnect turned on.
```

File: textile/chat-features.textile [hunk 2]
```diff
@@ -34,6 +34,36 @@ h3(#rooms-general). General Information
...
+** @(CHA-CS1c)@ The @CONNECTED@ status is used when the client connected to Ably servers.
+** @(CHA-CS1d)@ The @DISCONNECTED@ status is used when the client is not currently connected to Ably servers. This state may be temporary as the underlying Realtime SDK seeks to reconnect.
+** @(CHA-CS1e)@ The @SUSPENDED@ status is used when the client is in an extended state of disconnection, but will attempt to reconnect.
+** @(CHA-CS1f)@ The @FAILED@ status is used when the client is disconnected from the Ably servers due to some non-retriable failure such as authentication failure. It will not attempt to reconnect.
+* @(CHA-CS2)@ The connection must expose its current status.
+** @(CHA-CS2a)@ @[Testable]@ The chat client must expose its current connection status, a single value from the list in @CHA-CS1@.
+** @(CHA-CS2b)@ @[Testable]@ The chat client must expose the latest error, if any, associated with its current status.
+* @(CHA-CS3)@ @[Testable]@ The initial status and error of the connection will be whatever status the realtime client returns whilst the connection status object is constructed.
```
Comment (splindsay-92) [inline textile/chat-features.textile:51]:
```suggestion
* @(CHA-CS3)@ @[Testable]@ The initial status and error of the connection shall be whatever status the realtime client returns whilst the connection status object is constructed.
```

File: textile/chat-features.textile [hunk 3]
```diff
@@ -34,6 +34,36 @@ h3(#rooms-general). General Information
...
+*** @(CHA-CS5a1)@ @[Testable]@ If the realtime connection status transitions from @CONNECTED@ to @DISCONNECTED@, the chat client connection status must not change. A 5 second transient disconnect timer shall be started.
+*** @(CHA-CS5a2)@ @[Testable]@ If a transient disconnect timer is active and the realtime connection status changes to @DISCONNECTED@ or @CONNECTING@, the library must not emit a status change.
+*** @(CHA-CS5a3)@ @[Testable]@ If a transient disconnect timer is active and the realtime connections status changes to @CONNECTED@, @SUSPENDED@ or @FAILED@, the library shall cancel the transient disconnect timer. The superseding status change shall be emitted.
+*** @(CHA-CS5a4)@ @[Testable]@ If a transient disconnect timer expires the library shall emit a connection status change event. This event must contain the current status of of timer expiry, along with the @original@ error that initiated the transient disconnect timer.
```
Comment (splindsay-92) [inline textile/chat-features.textile:64]:
```suggestion
*** @(CHA-CS5a4)@ @[Testable]@ If a transient disconnect timer expires, the library shall emit a connection status change event. This event must contain the current status of timer expiry, along with the @original@ error that initiated the transient disconnect timer.
```

**Review by splindsay-92 — COMMENTED** [review]
Few small comments :)

File: textile/chat-features.textile (hunk 2 above)
Comment (AndyTWF) [reply textile/chat-features.textile]: Done!

**Review by splindsay-92 — APPROVED** [review]

**Pull request merged**